compiler/rustc_ast_lowering/src/asm.rs

```diff
@@ -374,11 +374,28 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                                         _ => None,
                                     };
 
+                                    let reg_str = |idx, default| -> &str {
+                                        let (op, _): &(InlineAsmOperand, Span) = &asm.operands[idx];
+                                        match op {
+                                            InlineAsmOperand::In { reg, .. }
+                                            | InlineAsmOperand::Out { reg, .. }
+                                            | InlineAsmOperand::InOut { reg, .. }
+                                            | InlineAsmOperand::SplitInOut { reg, .. } => {
+                                                if let InlineAsmRegOrRegClass::Reg(reg_sym) = reg {
+                                                    reg_sym.as_str()
+                                                } else {
+                                                    default
+                                                }
+                                            }
+                                            _ => default,
```

Comment:
```suggestion
                                            _ => unreachable!(),
```
I don't think the `default` is ever necessary here, both branches using `default` are unreachable.